src/lib/core/core.gni

```diff
@@ -54,6 +54,10 @@ if (chip_target_style == "") {
 if (chip_logging_style == "") {
   if (current_os == "android") {
     chip_logging_style = "android"
+  } else if (current_os == "ios") {
+    chip_logging_style = "external"
+  } else if (current_os == "mac" && current_toolchain == "//config/standalone/toolchain:standalone") {
```

Comment:
@mspang I really dislike this line but I didn't found the proper way to configure `chip_logging_style`to `external`without it. If I just use `current_os == "mac"` then some tests can not build anymore.

I think what's going on is that some of the tests relies on `src/lib/core` but they do not really uses the provided device layer. So then ends up not compiling the predefined `LogV`function from `src/lib/support` but there is not alternatives for `LogV` for them.

Do you have any suggestions about how to properly fix that ?

Reply:
Isn't this just saying

 `} else if (current_os == "mac" && !chip_build_tests) {
`

?

Reply:
This is tripping over a dependency cycle.

I'm told `platform` isn't what you'd expect from the name: platform in weave depend on core, **not the other way around**.

But all of this is totally broken right now.

What this change entails is basically that src/lib/support now depends on platform. The reverse dependency is already there. It's just more fallout from the cycle and strange layering structure.

You could add the code to `src/lib/support/mac` or something like that, or just add arrange things such that src/lib/support always brings along with it src/platform. It has to be so, because the deps are cyclic.

Reply:
Also re: platform being not in the expected place in the hierarchy,  you might notice src/platform isn't generally where all the platform code for low level CHIP components comes from - src/system, src/lib/support directly implement platform specifics. This includes IP networking, locking, timers, and other platform specific logging styles such as Android's. None of those come from the device layer (and couldn't if the unexpected stacking where platform is one of the topmost layers is applied consistently).

Reply:
Based on your comments, I change the code to make it works the same way the Android logging code works. Let me know if that sounds better to you.